tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -1892,6 +1903,277 @@ def test_get_all_datasets_200(self):
             ]
             self.assertEqual(expected_assets, dataset["assets"])
 
+        with self.subTest("Response Dataset objects contain visibility"):
+            for dataset in response.json:
+                self.assertEqual(Visibility.PUBLIC.name, dataset["visibility"])
+
+        with self.subTest("Response Dataset objects contain revision_of_collection"):
+            for dataset in response.json:
+                self.assertIsNone(dataset["revision_of_collection"])
+
+        with self.subTest("Response Dataset objects contain revision_of_dataset"):
+            for dataset in response.json:
+                self.assertIsNone(dataset["revision_of_dataset"])
+
+    def test_get_private_datasets_200(self):
+        """
+        Tests /curation/v1/datasets endpoint with different credentials and visibility filters.
+
+        Test data setup:
+        - published_collection_1: 2 datasets
+        - published_collection_2: 1 dataset, other owner
+        - published_collection_3: 1 dataset, previously revised and published
+        - unpublished_collection_1: 3 datasets
+        - unpublished_collection_2: 2 datasets, other owner
+        - published_collection_1_revision: 3 datasets (one new, one modified)
+
+        """
+        #
+        # Setup
+        #
+
+        # Public collection, 2 datasets.
+        published_collection_1 = self.generate_published_collection(
+            add_datasets=2,
+            metadata=CollectionMetadata(
+                "test_collection_1",
+                "described",
+                "john doe",
+                "john.doe@email.com",
+                [],
+                ["Consortia 1", "Consortia 2"],
+            ),
+        )
+
+        # Public collection, 2 datasets, other owner.
```

Comment:
1 dataset?

Reply:
I would just change the comment. I verified you were relying on published_collection_1 for testing unchanged datasets in revision mode.